docs/scopes.md

```diff
@@ -0,0 +1,179 @@
+and the rules for parsing and validating them.
+
+## Short-name scopes
+
+FxA supports a small set of "short-name" scopes
+that are identified by a short English word.
+Thees correspond either to
+scopes defined by external specifications (e.g. OpenID Connect),
+or to legacy scopes
+introduced during earily development.
+
+**No new short-name scopes should be added.**
+Instead we prefer to use URLs for new scopes,
+both to ensure uniqueness
+and to simplify parsing rules.
+
+Short-name scopes imply read-only access by default,
+with write access indicated by the suffix ":write".
+The may also have "sub-scopes"
+to indicate finer-grained access control.
+For example:
+
+* `profile` indicates read-only access
+  to the user's profile data.
+* `profile:write` indicates read/write access
+  to the user's profile data.
+* `profile:email` indicates read-only access
+  to the user's email address, but not any other
+  profile data.
+* `profile:email:write` indicates read/write access
+  to the user's email address.
+
+The following short-name scopes are recognized
+in the FxA ecosystem.
+
+### Profile data
+
+* `profile`: access the user's profile data.
+* `profile:uid`: access the user's opaque user id.
+* `profile:email`: access the user's email address.
+* `profile:locale`: access the user's locale.
+* `profile:avatar`: access the user's avatar picture.
+* `profile:display_name`: access the user's human-readable display name.
+* `profile:amr`: access information about the user's authentication methods and 2FA status.
+
+### OpenID Connect
+
+* `openid`: used to request an OpenID Connect `id_token`.
+* `email`: a synonym for `profile:email`, defined by the OIDC spec.
+
+### OAuth Client Management
+
+* `clients`: access the list of OAuth clients connected to a user's account.
+* `oauth`: register a new OAuth client record.
+
+### Basket
+
+* `basket`: access the user's subscription data in
+  [basket](http://basket.readthedocs.io/)
+
+
+## URL Scopes
+
+For new applications, scopes are identified by URLs.
+This helps to ensure uniqueness
+and reduces ambiguity in parsing.
+URL-based scopes imply read/write access by default,
+are compared as heirarchical resource references,
+and use the hash fragment for permission qualifiers.
+For example:
+
+* `https://identity.mozilla.com/apps/oldsync` indicates full
+   access to the user's data in Firefox Sync.
+* `https://identity.mozilla.com/apps/oldsync/bookmarks` indicates
+   full access to the user's bookmark data in Firefox Sync,
+   but not to other data types.
+* `https://identity.mozilla.com/apps/oldsync#read` indicates
+   read-only access to the user's data in Firefox Sync.
+* `https://identity.mozilla.com/apps/oldsync/history#write` indicates
+   write-only access to the user's history data in Firefox Sync.
+
+To be a valid scope, the URL must:
+
+* Be an absolute `https://` URL.
+* Have no query component.
+* Have a fragment component consisting only of alphanumeric ascii characters.
```

Comment:
Random drive-by comment, but this wording makes it sound like the fragment part is mandatory because of the MUST before the list. Maybe clarify here with an "if specified..." clause, or split out to a separate MAY list?

Reply:
(or include MUST / MAY in each bullet rather than aggregated at the top)

Reply:
Thanks, good call, I will update.